File: Basic Coding Problems/01 Fundamentals/001 Basic I/Java/DataTypesInputOutput.java
import java.util.Scanner;

public class DataTypesInputOutput {
    public static void main(String[] args) {
        byte byteValue;
        char character;
        short shortInteger;
        int integer;
        long longInteger;
        float floatValue;
        double doubleValue;
        boolean booleanValue;
        String stringValue;

        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a byte: ");
        byteValue = scanner.nextByte();

        System.out.print("Enter a character: ");
        character = scanner.next().charAt(0);

        System.out.print("Enter a short integer: ");
        shortInteger = scanner.nextShort();

        System.out.print("Enter an integer: ");
        integer = scanner.nextInt();

        System.out.print("Enter a long integer: ");
        longInteger = scanner.nextLong();

        System.out.print("Enter a float: ");
        floatValue = scanner.nextFloat();

        System.out.print("Enter a double: ");
        doubleValue = scanner.nextDouble();

        System.out.print("Enter a boolean: ");
        booleanValue = scanner.nextBoolean();

        System.out.print("Enter a string: ");
        stringValue = scanner.next();

        System.out.println("\nThe byte is: " + byteValue);
        System.out.println("\nThe character is: " + character);
        System.out.println("The short integer is: " + shortInteger);
        System.out.println("The integer is: " + integer);
        System.out.println("The long integer is: " + longInteger);
        System.out.println("The float is: " + floatValue);
        System.out.println("The double is: " + doubleValue);
        System.out.println("The boolean is: " + booleanValue);
        System.out.println("The string is: " + stringValue);

        scanner.close();
    }
}

File: Basic Coding Problems/01 Fundamentals/001 Basic I/C++/DataTypesInputOutput.cpp
#include <iostream>

using namespace std;

int main() {
    char character;
    unsigned char unsignedCharacter;
    short shortInteger;
    int integer;
    unsigned int unsignedInteger;
    long longInteger;
    unsigned long unsignedLong;
    unsigned long long unsignedLongLong;
    float floatValue;
    double doubleValue;
    long double longDoubleValue;
    string stringVariable;

    cout << "Enter a character: ";
    character = cin.get();
    cin.ignore();

    cout << "Enter an unsigned character: ";
    unsignedCharacter = cin.get();
    cin.ignore();

    cout << "Enter a short integer: ";
    cin >> shortInteger;

    cout << "Enter an integer: ";
    cin >> integer;

    cout << "Enter an unsigned integer: ";
    cin >> unsignedInteger;

    cout << "Enter a long integer: ";
    cin >> longInteger;

    cout << "Enter an unsigned long integer: ";
    cin >> unsignedLong;

    cout << "Enter an unsigned long long integer: ";
    cin >> unsignedLongLong;

    cout << "Enter a float: ";
    cin >> floatValue;

    cout << "Enter a double: ";
    cin >> doubleValue;

    cout << "Enter a long double: ";
    cin >> longDoubleValue;

    cout << "Enter a string: ";
    cin.ignore();
    getline(cin, stringVariable);

    cout << "\nThe character is: " << character << endl;
    cout << "The unsigned character is: " << unsignedCharacter << endl;
    cout << "The short integer is: " << shortInteger << endl;
    cout << "The integer is: " << integer << endl;
    cout << "The unsigned integer is: " << unsignedInteger << endl;
    cout << "The long integer is: " << longInteger << endl;
    cout << "The unsigned long integer is: " << unsignedLong << endl;
    cout << "The unsigned long long integer is: " << unsignedLongLong << endl;
    cout << "The float is: " << floatValue << endl;
    cout << "The double is: " << doubleValue << endl;
    cout << "The long double is: " << longDoubleValue << endl;
    cout << "The string is: " << stringVariable << endl;

    return 0;
}